File: rust/src/desing_patters/repository.rs
pub trait UserRepository {
    fn find(&self) -> &[String];
    fn save(&mut self, date: String);
}

pub struct ImplRepository {
    list: Vec<String>,
}

impl ImplRepository {
    pub fn new() -> ImplRepository {
        return ImplRepository { list: Vec::new() };
    }
}

impl UserRepository for ImplRepository {
    fn save(&mut self, date: String) {
        self.list.push(date)
    }
    fn find(&self) -> &[String] {
        return &self.list;
    }
}

File: Go/designPatterns/repository/repository.go
package repository

type Repository interface {
	Save(dato string)
	Find() []string
}

type ImplRepository struct {
	list []string
}

func New() *ImplRepository {
	return &ImplRepository{}
}

func (r *ImplRepository) Save(dato string) {
	r.list = append(r.list, dato)
}

func (r *ImplRepository) Find() []string {
	return r.list
}
